### ISSUE TITLE
Problem Bingding with VNI Telex

### ISSUE THREAD

**Issue by phanthanhdat077** [dotnet/wpf#9248]
I can not binding to Text from TextBox my Text property always empty when I using VIE Vietnamese Telex from Keyboard layout Win 11
![image](https://github.com/dotnet/wpf/assets/34691283/0179fdf4-d997-4945-8806-f798de862e43)

![image](https://github.com/dotnet/wpf/assets/34691283/6c6e66bf-01d3-44c1-a580-337c0556b2ee)

![image](https://github.com/dotnet/wpf/assets/34691283/1722e1ed-dbeb-4d93-84f5-30d14cafa5e0)

[WpfApp14.zip](https://github.com/user-attachments/files/15836121/WpfApp14.zip)

**Comment by lindexi** [comment]
@phanthanhdat077 Can it work well with other IME?

**Comment by lindexi** [comment]
@phanthanhdat077 Sorry, I don't know anything about Vietnamese Telex right now

**Comment by miloush** [comment]
If it helps to investigate, note that Vietnamese Telex is IME. The view model property is not getting updated until the composition is committed. It does seem to be reflected in the TextBox.Text property though on every TextChanged event.

**Comment by lindexi** [comment]
@miloush What I'm confused about is why is the TextChanged event triggered but the Text is empty.

**Comment by miloush** [comment]
As I said, the databinding does not update the Text property. The Text is only empty on the viewmodel, not on the TextBox. I would guess that the composition events do not change the TextBox.Text property in a way that a databinding with PropertyChanged trigger would pick it up.

**Comment by phanthanhdat077** [comment]
> As I said, the databinding does not update the Text property. The Text is only empty on the viewmodel, not on the TextBox. I would guess that the composition events do not change the TextBox.Text property in a way that a databinding with PropertyChanged trigger would pick it up.

yes TextBox.Text still received text I just can not binding to viewmodel.

**Comment by miloush** [comment]
Your viewmodel will get the text once the input is committed, e.g. after typing a space. This applies to all similar IMEs, not only Vietnamese Telex. It does feel like a bug that the TextBox is firing TextChanged but not the DP value change. I am a bit busy at the moment to have a closer look, perhaps others can give it a go.

**Comment by phanthanhdat077** [comment]
my problem: I using Textbox for typing only numbers and can't space that method for payment it has the event TxtInputMoney_PreviewTextInput to handle for type only numbers but when I turn on Vietnamese Telex it can overcome the event TxtInputMoney_PreviewTextInput  and type the word.

private void TxtInputMoney_PreviewTextInput(object sender, TextCompositionEventArgs e)
{
    Regex regex = new Regex("[^0-9]+");
    e.Handled = regex.IsMatch(e.Text);
}
